docs/cpp/guides/flags.md

```diff
@@ -410,3 +422,131 @@ Abseil flags library to suppress normal error signaling that occurs when
 `--name` is seen on the command-line (or `--noname` since a listed flag might
 have been an old boolean flag), but `name` has not been defined anywhere in the
 application
+
+## Defining Custom Flag Types {#custom}
+
+For a type `T` to be used as an Abseil flag type, it must support conversion to
+and from strings supplied on the command-line. Custom types may have a unique
+format for this command-line string, and hence may require custom support for
+Abseil flags.
+
+To add support for your user-defined type, add overloads of `AbslParseFlag()`
+and `AbslUnparseFlag()` as free (non-member) functions to your type. If `T`
+is a class type, these functions can be friend function definitions. These
+overloads must be added to the same namespace where the type is defined, so
+that they can be discovered by Argument-Dependent Lookup (ADL).
+
+Example:
+
+```cpp
+namespace foo {
+enum OutputMode { kPlainText, kHtml };
+
+// AbslParseFlag converts from a string to OutputMode.
+// Must be in same namespace as OutputMode.
+
+// Parses an OutputMode from the command line flag value `text. Returns
+// `true` and sets `*mode` on success; returns `false` and sets `*error`
+// on failure.
+bool AbslParseFlag(absl::string_view text,
+                   OutputMode* mode,
+                   std::string* error) {
+  if (text == "plaintext") {
+    *mode = kPlainText;
+    return true;
+  }
+  if (text == "html") {
+    *mode = kHtml;
+    return true;
+  }
+  *error = "unknown value for enumeration";
+  return false;
+}
+
+// AbslUnparseFlag converts from an OutputMode to a string.
+// Must be in same namespace as OutputMode.
+
+// Returns a textual flag value corresponding to the OutputMode `mode`.
+std::string AbslUnparseFlag(OutputMode mode) {
+  switch (mode) {
+   case kPlainText: return "plaintext";
+   case kHtml: return "html";
+   default: return SimpleItoa(mode);
```

Comment:
s/SimpleItoa/absl::StrCat/ ?

Reply:
Fixed in source. Will push at next doc push.